src/telem/lib/input/mekanism/LaserAmplifierInputAdapter.lua

```diff
@@ -0,0 +1,67 @@
+local o = require 'telem.lib.ObjectModel'
+local t = require 'telem.lib.util'
+
+local InputAdapter      = require 'telem.lib.InputAdapter'
+local Metric            = require 'telem.lib.Metric'
+local MetricCollection  = require 'telem.lib.MetricCollection'
+
+local LaserAmplifierInputAdapter = o.class(InputAdapter)
+LaserAmplifierInputAdapter.type = 'LaserAmplifierInputAdapter'
+
+function LaserAmplifierInputAdapter:constructor (peripheralName, categories)
+    self:super('constructor')
+
+    -- TODO this will be a configurable feature later
+    self.prefix = 'meklaseramp:'
+
+    -- TODO make these constants
+    local allCategories = {
+        'basic',
+    }
+
+    if not categories then
+        self.categories = { 'basic' }
+    elseif categories == '*' then
+        self.categories = allCategories
+    else
+        self.categories = categories
+    end
+
+    -- boot components
+    self:setBoot(function ()
+        self.components = {}
+
+        self:addComponentByPeripheralID(peripheralName)
+    end)()
+end
+
+function LaserAmplifierInputAdapter:read ()
+    self:boot()
+
+    local source, laser = next(self.components)
+
+    local metrics = MetricCollection()
+
+    local loaded = {}
+
+    for _,v in ipairs(self.categories) do
+        -- skip, already loaded
+        if loaded[v] then
+            -- do nothing
+
+        -- Literally all we have lmao
+        elseif v == 'basic' then
```

Comment:
Rename `energy_max` to `max_energy` to align with induction matrix and turbine adapters. I would also add an energy category:

Basic:
- energy_filled_percentage

Energy:
- energy
- max_energy
- energy_needed

The energy category across most machines is nearly identical... I may try to look into ways of applying metric "templates" for common categories in the future to improve code reuse.